pkg/governance-scripts/contracts/TimelockAuthorizerTransitionMigrator.sol

```diff
@@ -97,6 +112,26 @@ contract TimelockAuthorizerTransitionMigrator {
         );
     }
 
+    /**
+     * @notice Executes permissions scheduled during migration, all at once.
+     * @dev `migratePermissions` must be called successfully first.
+     * Emits `ScheduledPermissionSkipped` event for permission IDs that cannot be executed at the time for any reason
+     * (delay not yet due, action was canceled, or action already executed).
+     */
```

Comment:
Maybe add here why it's ok to call this multiple times (more explicitly). As I understand it, if you had 3 things with a week delay, and 3 with a 2-week delay, and you called this after a week, it would grant permissions on the 3 1-week ids, and skip the 2-week ones.

Then if you waited another week and called it again, it would skip the first three - canExecute would return false because it "executed" would have been set to true - and grant the last 3.

So I think you don't need that "ScheduledPermissionSkipped" event. Events usually indicate either a successful action, or an error. But here it would be emitted as part of normal operation (unless all permissions happened to be the same length, and you only called it once). You could call this forever after, and it would just keep emitting skipped events.

Reply:
+1, I'll update the docs.

Regarding the event, your reasoning is correct.

The intention behind it was to be able to reason about pending permissions by quickly looking at the transaction events. Without the event, if a permission or two is skipped on the first call, it would be a bit more difficult to tell whether the migration is complete or not, since the timelock authorizer will only emit an event when granting a permission in this case. That being said, do you still think we should remove it?

Reply:
It's not the end of the world, but it seems tracking the "positive" events should be sufficient to know whether permissions have been granted. I don't particularly like having a function anybody can call forever that would emit a bunch of events. A bot could flood the subgraph with spurious events.

Reply:
Sounds good then.

Removed in https://github.com/balancer-labs/balancer-v2-monorepo/pull/2184/commits/fe318378668813bd162da10b5aa94c1422530fa2.